### PR TITLE
Fix toolbar keyboard navigation after pressing alt+F10

### LINKED ISSUE #28016: Moving through block toolbar after pressing alt+F10 hides the toolbar
**Describe the bug**

After pressing <kbd>alt+F10</kbd> to focus on the block toolbar, moving through the toolbar items with arrow keys will unexpectedly hide it.

**To reproduce**
1. Create a new post
2. Create a paragraph and write some text
3. With the block toolbar hidden, press <kbd>alt+F10</kbd>
4. Move through the toolbar items using arrow keys

**Expected behavior**

The toolbar shouldn't hide when using arrow keys.

**Screenshots**

Using the block toolbar

**Editor version (please complete the following information):**

master

**Desktop (please complete the following information):**
 - OS: macOS Big Sur 11.1
 - Browser: Chrome
 - Version: 87

### PR DESCRIPTION
**Pull request by diegohaz**
Closes #28016

The specific block's toolbar items (`Change text alignment`, `Bold` etc. for Paragraph) are added through the `SlotFill` module, which uses React Portal. The block popover has a `Slot` component, whose node serves as the container for the portal. React events only propagate when elements are in the same React subtree, which doesn't happen in this case.

When those elements receive focus, React's `onFocus` and `onBlur` will not propagate, so the popover thinks they're outside. In fact, those elements are outside of the popover in the React tree, so this makes sense.

~Had to convert those events to DOM ones. But I'm open to suggestions. I thought about changing it directly on the `useFocusOutside` hook, but it would be more complicated since this is a public API.~

This PR removes the `onFocusOutside` behavior and uses a logic similar to when we press <kbd>Shift+Tab</kbd>.

## Screenshots

Navigating through the block toolbar after pressing alt+F10

## How has this been tested?

- Check the added e2e test.
- Use the block toolbar and check if #28016 is fixed and the toolbar behaves correctly.

## Types of changes

Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by diegohaz** [comment]
This is marked as a regression because the issue didn't exist before #27043, but I think it's just because [this `onBlur`](https://github.com/WordPress/gutenberg/pull/27043/files#diff-81b4cbf56c184d05760b9427d377b771891488d40e297f741922b6bf6d422e3bL183-R187) was just being ignored and the toolbar wasn't being hidden after pressing alt+F10 whatsoever.

File: packages/components/src/popover/index.js [hunk 1]
```diff
@@ -199,6 +200,22 @@ function setClass( element, name, toggle ) {
 	}
 }
 
+function useFocusOutsideWithRef( ref, onFocusOutside ) {
```
Comment (youknowriad) [inline packages/components/src/popover/index.js:203]: To be honest,  I feel useFocusOutside should use Dom events by default, that said, when we tried that recently, we had an issue with "nested popovers" and "nested modals" and we didn't really found a way to solve it without DOM events.

Comment (youknowriad) [reply packages/components/src/popover/index.js:203]: I believe the change in this PR is probably going to break some of these "nested popovers"  behaviors.

Comment (diegohaz) [reply packages/components/src/popover/index.js]: @youknowriad Do you know which part of the code is responsible for focusing on the toolbar when pressing shift + tab? I'm curious why the toolbar behavior isn't affected in that case, only when using alt + F10.

Comment (youknowriad) [reply packages/components/src/popover/index.js]: Not on top of my head, maybe @ellatrix knows

**Comment by diegohaz** [comment]
As @youknowriad pointed out, using DOM events on `useFocusOutside` produced side effects that affected nested popovers. I couldn't find a way to make it work with this approach.

The solution I found was getting rid of the problematic `onFocusOutside` prop and use `stopTyping` so the block popover is shown as when we press <kbd>Shift+Tab</kbd>.

Maybe there's a better way to do that (maybe I shouldn't use `stopTyping` directly?). I don't know. But the issue seems to be fixed. I'd appreciate a review. ❤️

File: packages/components/src/popover/index.js (hunk 1 above)
Comment (ellatrix) [reply packages/components/src/popover/index.js]: That should be writing flow => focus capture elements. Focus is transferred to the focus capture element and then the browser focusses the next element in the DOM.

**Comment by youknowriad** [comment]
@diegohaz I'm actually unable to reproduce the bug  in master, maybe it was already fixed by something else?

**Comment by diegohaz** [comment]
@youknowriad I can still reproduce it on master. Did you follow the exact instructions (the toolbar must be hidden and should be focused with <kbd>alt+F10</kbd>)?

**Comment by youknowriad** [comment]
I think I missed the detail about  the isTyping needing to be true. 👍

**Review by youknowriad — APPROVED** [review]
This works well for me. Thanks @diegohaz

**Pull request merged**